src/Repositories/FilesystemRepository.php

```diff
@@ -43,6 +43,31 @@ public function get(string $path): ?Sheet
         });
     }
 
+    public function getBySlug(string $slug): ?Sheet
+    {
+        $parts = explode('/', $slug);
+
+        $name = array_pop($parts);
+
+        $pattern = sprintf(
```

Comment:
Kinda wondering whether this pattern should be moved to the slugparser classes.

Reply:
Yeah, this PR isn't perfect right now.
Have to handle the "does not exist case" - right now the `get()` method throws an unexpected type exception.

The pattern could be better/more precise if done in the path parser. But the question is if we move the whole pattern or only the a special part?
The whole would mean a bit duplication for the two default ones but would allow much more powerful userland ones.
It's also a question if the parser provides a `buildPattern()` with some inputs - or returns a static pattern with placeholders?

So this PR was more like a first discussion base.

Reply:
Got it! I like the idea of having unlimited options in userland, so I kind of favor moving the entire pattern to the parsers. 

I think having the parser return the contents of `$pattern` should suffice. What do you think?